src/datepicker/datepicker.js

```diff
@@ -629,7 +629,17 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
         scope.$parent.$watch(getAttribute, function(value) {
           scope.watchData[key] = value;
           if (key === 'minDate' || key === 'maxDate') {
-            cache[key] = new Date(value);
+            var date = parseDate(value);
+            if (!date) {
+              date = new Date(value);
+              if (angular.isDate(date) && !isNaN(date)) {
+                var dateArray = date.toJSON().split('T')[0].split('-');
+                date = new Date(+dateArray[0], +dateArray[1] - 1, +dateArray[2]);
```

Comment:
I should have caught this earlier, but it would be better to do `date = new Date(date)` - this preserves the timezone of `date`, in the event that the value is a date object.